deepspeed/runtime/utils.py

```diff
@@ -937,38 +930,17 @@ def align_dense_tensors(tensor_list, alignment):
     return padded_tensor_list
 
 
-def all_gather_dp_groups(partitioned_param_groups, dp_process_group, start_alignment_factor, allgather_bucket_size):
```

Comment:
> @tjruwase do we still want to maintain the original all-gather functionality with `nccl_start_alignment_factor` like this https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/runtime/utils.py#L955 ?

I think nccl_start_alignment_factor  is only needed when doing allgather in shards

Reply:
@GuanhuaWang, yes let's maintain for backwards-compatibility.

Reply:
> @GuanhuaWang, yes let's maintain for backwards-compatibility.

Hi @taozhiwei 

I am aligned here with @tjruwase , instead of delete older version, could you keep both your new version and older one? thanks

Reply:
> > @GuanhuaWang, yes let's maintain for backwards-compatibility.
> 
> Hi @taozhiwei
> 
> I am aligned here with @tjruwase , instead of delete older version, could you keep both your new version and older one? thanks

I have already pushed the backward compatible code. could you please help me review it.thanks

Reply:
> > @GuanhuaWang, yes let's maintain for backwards-compatibility.
> 
> Hi @taozhiwei
> 
> I am aligned here with @tjruwase , instead of delete older version, could you keep both your new version and older one? thanks

@GuanhuaWang I have already pushed the backward compatible code. could you please help me review it.thanks